src/ast/desugar.rs

```diff
@@ -118,7 +118,10 @@ fn add_semi_naive_rule(desugar: &mut Desugar, rule: Rule) -> Option<Rule> {
     let mut var_set = HashSet::default();
     for head_slice in new_rule.head.0.iter_mut().rev() {
         match head_slice {
-            Action::Set(span, _, _, expr) => {
+            Action::Set(span, _, _, expr, is_cost) => {
```

Comment:
Does cost depend only on the inputs of a function right now? Should we only allow setting costs for constructors, not functions?

Reply:
Yes cost depends only on the inputs, similar to subsumed.